external/storm-eventhubs/src/main/java/org/apache/storm/eventhubs/spout/EventHubReceiverImpl.java

```diff
@@ -65,24 +69,38 @@ public EventHubReceiverImpl(EventHubSpoutConfig config, String partitionId) {
   }
 
   @Override
-  public void open(IEventHubFilter filter) throws EventHubException {
-    logger.info("creating eventhub receiver: partitionId=" + partitionId + 
-    		", filterString=" + filter.getFilterString());
+  public void open(String offset) throws EventHubException {
+    logger.info("creating eventhub receiver: partitionId=" + partitionId +
+            ", offset=" + offset);
     long start = System.currentTimeMillis();
-    receiver = new ResilientEventHubReceiver(connectionString, entityName,
-    		partitionId, consumerGroupName, defaultCredits, filter);
-    receiver.initialize();
-    
+    try {
+      ehClient = EventHubClient.createFromConnectionString(connectionString).get();
+      receiver = ehClient.createEpochReceiver(
+              consumerGroupName,
+              partitionId,
+              offset,
+              false,
+              1).get();
+    }catch (Exception e){
+      logger.info("Exception in creating EventhubClient"+e.toString());
+    }
     long end = System.currentTimeMillis();
     logger.info("created eventhub receiver, time taken(ms): " + (end-start));
   }
 
   @Override
-  public void close() {
+  public void close(){
     if(receiver != null) {
-      receiver.close();
+      try {
+        receiver.close().get();
```

Comment:
>.get() [](start = 24, length = 6)

if receiver.close() fails - this results into connection leak; handle this case. something like:
receiver.whenComplete(() -> { ehClient.closeSync() }).get();